go/arrow/float16/float16.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package float16
+
+import (
+	"math"
+)
+
+type Float16 uint16
```

Comment:
actually, independantly of how one would call that type, I would perhaps change its definition to:
```go
type Float16 struct { v uint16 }
```

to prevent mistakes such as:
```go
var v float16.Float16 = 3 // oops... this compiles. it shouldn't.
```
https://play.golang.org/p/91qGe456FPA